magma/backend/mlir/sv.py

```diff
@@ -286,13 +296,17 @@ def print(self, printer: PrinterBase, opts: PrintOpts):
         printer.print("}")
         if self._else_block is None:
             printer.flush()
+            if opts.print_locations:
+                printer.print_line(self.location.emit())
             return
         printer.print(" else {")
         printer.flush()
         printer.push()
         self._else_block.print(printer, opts)
         printer.pop()
         printer.print_line("}")
+        if opts.print_locations:
+            printer.print_line(self.location.emit())
```

Comment:
Seems like we always check `opts` and print after the function returns? Perhaps we can just wrap these in a decorator that does that at the end of the function (rather than adding the code at the various return points).